Mathlib/AlgebraicTopology/SimplexCategory.lean

```diff
@@ -55,7 +63,7 @@ def mk (n : ℕ) : SimplexCategory :=
   n
 
 /-- the `n`-dimensional simplex can be denoted `[n]` -/
-scoped[Simplicial] notation "[" n "]" => SimplexCategory.mk n
+scoped[Simplicial] notation (priority := high) "[" n "]" => SimplexCategory.mk n
```

Comment:
This change would also allow removing many of the `([n] : SimplexCategory)` type annotations in this file and elsewhere (though maybe this suggests there is a reason it hasn't been done before).

I can open a separate PR removing obsolete type annotations.

Reply:
I have little understanding of the detailed syntax of notations and of meta code, but `[...]` may conflict with list notations. However, as a scoped `Simplicial` notation, we can certainly increase the priority.

Reply:
Unfortunately `scoped[Simplicial] syntax` is not possible (unless #18230 is merged), so currently `open SimplexCategory.Truncated` brings the `[m]ₙ` notation into scope. Do you think this notation also being given `(priority := high)` could cause any problems?

Reply:
hopefully #18230 is compiling soon :)

Reply:
#21565 changed the notation for objects of the simplex category, and I have done the same for the truncated simplex category notation in this PR. The new syntax allows removing `(priority := high)`.

I still think it would be a good idea to make this notation `scoped[Simplicial]` after #18230 lands.